ui/app/src/components/GitManagement/GitManagement.tsx

```diff
@@ -121,22 +123,46 @@ export function GitManagement(props: GitManagementProps) {
     },
     [setRepoStatusState]
   );
-  const fetchRepoStatus = useCallback(() => {
+  const fetchRepoStatus = useCallback((): Promise<RepositoryStatus> => {
     setRepoStatusState({ loadingStatus: true, statusFetchError: null });
-    fetchStatus(siteId).subscribe({
-      next: fetchRepoStatusReceiver,
-      error(response) {
-        setRepoStatusState({ loadingStatus: false, statusFetchError: response });
-        dispatch(
-          showSystemNotification({
-            message: response.response.message,
-            options: { variant: 'error' }
-          })
-        );
-      }
+    return new Promise((resolve, reject) => {
+      fetchStatus(siteId).subscribe({
+        next: (repoStatus) => {
+          fetchRepoStatusReceiver(repoStatus);
+          resolve(repoStatus);
+        },
+        error(response) {
+          setRepoStatusState({ loadingStatus: false, statusFetchError: response });
+          dispatch(
+            showSystemNotification({
+              message: response.response.message,
+              options: { variant: 'error' }
+            })
+          );
+          reject(response);
+        }
+      });
     });
   }, [dispatch, fetchRepoStatusReceiver, setRepoStatusState, siteId]);
 
+  const onPullSuccess = (result: PullResponse) => {
+    fetchRepoStatus();
+  };
+
+  const onPullError = (response) => {
```

Comment:
When the pull results in conflicts, it throws? Why is the pull error opening the dialog?

Reply:
If there are conflicts, pullError triggers, so when there are conflicts, after doing the fetchStatus (that fetch was already there), **if** there are conflicts, then the dialog will open